graphql/resolvers.go

```diff
@@ -194,6 +194,37 @@ func (r *mutationResolver) RemoveFavoriteProject(ctx context.Context, identifier
 	}, nil
 }
 
+func (r *mutationResolver) SpawnVolume(ctx context.Context, spawnVolumeInput SpawnVolumeInput) (bool, error) {
+	volume := GetVolumeFromSpawnVolumeInput(spawnVolumeInput)
+	success, _, gqlErr, err, vol := RequestNewVolume(ctx, volume)
+	if err != nil {
+		return false, gqlErr.Send(ctx, err.Error())
+	}
+	errorTemplate := "Volume %s has been created but an error occured."
+	if spawnVolumeInput.Expiration != nil && spawnVolumeInput.NoExpiration != nil && *spawnVolumeInput.NoExpiration == true {
```

Comment:
Does this validation need to come before requesting a new volume? Also would it make more sense to move them into the RequestNewVolume function?

Reply:
_Does this validation need to come before requesting a new volume?_
ah good point. things will be easier for the user if we don't create a volume when the input is invalid. 

_Also would it make more sense to move them into the RequestNewVolume function?_
I think the logic should stay in the resolver because RequestNewVolume is also used in service/spawn and doesn't require expiration options during volume creation.